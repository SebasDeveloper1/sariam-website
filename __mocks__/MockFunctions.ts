import '@testing-library/jest-dom/extend-expect';
export const MockFunction = jest.fn();
