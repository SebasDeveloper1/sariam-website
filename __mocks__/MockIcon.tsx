// Mock the startIcon and endIcon components for testing
export const MockIcon = (): JSX.Element => <div id="mock-icon">Icon</div>;
