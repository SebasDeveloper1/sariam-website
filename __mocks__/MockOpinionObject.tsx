import { MockOpinionTestProps } from '@/components/OpinionCard/OpinionCard.model'; // Mocked opinion data for testing
export const MockOpinion: MockOpinionTestProps = {
  author: {
    name: 'John Doe',
    role: 'CEO',
    photo: {
      url: 'https://example.com/photo.jpg',
      description: 'John Doe photo',
    },
  },
  content: 'This is a great product!',
  sys: { id: 'HUWhn2yQFFqvsAKxX6Htl' },
};
