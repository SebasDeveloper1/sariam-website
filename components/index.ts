import Button from './Buttons/Button/Button';
import IconButton from './Buttons/IconButton/IconButton';
import SliderComponent from './SliderComponent/SliderComponent';
import ServiceCard from './ServiceCard/ServiceCard';
export { Button, IconButton, SliderComponent, ServiceCard };
