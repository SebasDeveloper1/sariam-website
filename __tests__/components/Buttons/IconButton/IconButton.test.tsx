import { render, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { IconButton } from '@/components';
import { MockIcon } from '@/__mocks__/MockIcon';
import { MockFunction } from '@/__mocks__/MockFunctions';

describe('IconButton component', () => {
  describe('Rendering', () => {
    it('should render a button with default props', () => {
      const { getByRole } = render(<IconButton />);
      const button = getByRole('button');

      expect(button).toBeInTheDocument();
      expect(button).toHaveClass('button-primary');
      expect(button).not.toBeDisabled();
      expect(button).toHaveTextContent('');
      expect(document.querySelector('.text-3xl')).toBeNull();
    });

    it('should render a button with a custom variant', () => {
      const { getByRole } = render(<IconButton variant="secondary" />);
      const button = getByRole('button');

      expect(button).toBeInTheDocument();
      expect(button).toHaveClass('button-secondary');
      expect(button).not.toBeDisabled();
    });

    it('should render a button with custom styles', () => {
      const { getByRole } = render(<IconButton styles="bg-red-500" />);
      const button = getByRole('button');

      expect(button).toBeInTheDocument();
      expect(button).toHaveClass('bg-red-500');
    });

    it('should render a button with an icon', () => {
      const { getByRole } = render(<IconButton icon={MockIcon} />);
      const button = getByRole('button');

      expect(button).toBeInTheDocument();
      expect(button).toContainElement(document.querySelector('#mock-icon'));
    });

    it('should render a disabled button', () => {
      const { getByRole } = render(<IconButton disabled />);
      const button = getByRole('button');

      expect(button).toBeInTheDocument();
      expect(button).toBeDisabled();
    });

    it('should pass additional props to the button', () => {
      const { getByTestId } = render(<IconButton data-testid="test-button" />);
      const button = getByTestId('test-button');

      expect(button).toBeInTheDocument();
    });
  });

  describe('Interaction', () => {
    it('should call the custom onClick function once', () => {
      const { getByRole } = render(<IconButton onClick={MockFunction} />);
      const button = getByRole('button');

      expect(button).toBeInTheDocument();
      fireEvent.click(button);

      expect(MockFunction).toHaveBeenCalledTimes(1);
    });
  });

  describe('Snapshot', () => {
    it('should match the IconButton component snapshot', () => {
      const { asFragment } = render(<IconButton variant="secondary" />);

      expect(asFragment()).toMatchSnapshot();
    });
  });
});
