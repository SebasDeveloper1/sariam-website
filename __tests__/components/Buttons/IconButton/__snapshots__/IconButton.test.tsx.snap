// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton component Snapshot should match the IconButton component snapshot 1`] = `
<DocumentFragment>
  <button
    class="button-secondary rounded-full p-2.5  "
  />
</DocumentFragment>
`;

exports[`Render IconButton component Button component matches snapshot 1`] = `
<DocumentFragment>
  <button
    class="button-secondary rounded-full p-2.5  "
  />
</DocumentFragment>
`;
